### ISSUE TITLE
Rule Change: Implement suggestions for `no-empty-static-block`

### ISSUE THREAD

**Issue by jaymarvelz** [eslint/eslint#20054]
### What rule do you want to change?

no-empty-static-block

### What change do you want to make?

Implement suggestions

### How do you think the change should be implemented?

Other

### Example code

```js
/*eslint no-empty-static-block: "error"*/

class Foo {
    static {}
}
```

### What does the rule currently do for this code?

The `no-empty-static-block` rule has two inconsistencies compared to the `no-empty` rule:

1. Doesn't provide fixable suggestions to add comments
2. Highlights the entire static block instead of just the curly braces

### What will the rule do after it's changed?

After the fix, `no-empty-static-block` should behave consistently with `no-empty`:

1. Provide a suggestion to add `/* empty */` comment inside the static block
2. Highlight only the opening and closing braces `{}` instead of the whole static block

### Participation

- [x] I am willing to submit a pull request to implement this change.

**Comment by mdjermanovic** [comment]
Both changes make sense to me 👍 

@eslint/eslint-team thoughts?

**Comment by nzakas** [comment]
Sounds good. Marking as accepted.
